sqlglot/dialects/presto.py

```diff
@@ -186,6 +186,12 @@ def _unix_to_time_sql(self: Presto.Generator, expression: exp.UnixToTime) -> str
     return ""
 
 
+def _to_int(expression: exp.Expression) -> exp.Expression:
+    if expression.type and expression.type.this not in exp.DataType.INTEGER_TYPES:
```

Comment:
not sure if we should do this if no type is set.

Reply:
perhaps you can check if a type is set and if not then call annotate types on the `expression` subtree? see https://github.com/tobymao/sqlglot/blob/main/sqlglot/dialects/snowflake.py#L595

Reply:
ah, cool, sure